akka-cluster/src/multi-jvm/scala/akka/cluster/DowningWhenOtherHasQuarantinedThisActorSystemSpec.scala

```diff
@@ -0,0 +1,126 @@
+class DowningWhenOtherHasQuarantinedThisActorSystemMultiJvmNode2
+    extends DowningWhenOtherHasQuarantinedThisActorSystemSpec
+class DowningWhenOtherHasQuarantinedThisActorSystemMultiJvmNode3
+    extends DowningWhenOtherHasQuarantinedThisActorSystemSpec
+
+abstract class DowningWhenOtherHasQuarantinedThisActorSystemSpec
+    extends MultiNodeSpec(DowningWhenOtherHasQuarantinedThisActorSystemSpec)
+    with MultiNodeClusterSpec {
+  import DowningWhenOtherHasQuarantinedThisActorSystemSpec._
+
+  "Cluster node downed by other" must {
+
+    if (!ArterySettings(system.settings.config.getConfig("akka.remote.artery")).Enabled) {
+      // this feature only works in Artery, because classic remoting will not accept connections from
+      // a quarantined node, and that is too high risk of introducing regressions if changing that
+      pending
+    }
+
+    "join cluster" taggedAs LongRunningTest in {
+      awaitClusterUp(first, second, third)
+      enterBarrier("after-1")
+    }
+
+    "down itself" taggedAs LongRunningTest in {
+      runOn(first) {
+        testConductor.blackhole(first, second, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(third, second, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("blackhole")
+
+      within(15.seconds) {
+        runOn(first) {
+          awaitAssert {
+            cluster.state.unreachable.map(_.address) should ===(Set(address(second)))
+          }
+          awaitAssert {
+            // second downed and removed
+            cluster.state.members.map(_.address) should ===(Set(address(first), address(third)))
+          }
+        }
+        runOn(second) {
+          awaitAssert {
+            cluster.state.unreachable.map(_.address) should ===(Set(address(first), address(third)))
+          }
+        }
+      }
+      enterBarrier("down-second")
+
+      runOn(first) {
+        testConductor.passThrough(first, second, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(third, second, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("pass-through")
+
+      runOn(second) {
+        // shutting down itself
+        awaitCond(cluster.isTerminated)
+      }
+
+      enterBarrier("after-2")
+    }
+
+    "not be triggered by another node shutting down" taggedAs LongRunningTest in {
+      runOn(first) {
+        system.eventStream.subscribe(testActor, classOf[ThisActorSystemQuarantinedEvent])
+      }
+      enterBarrier("subscribing")
+
+      runOn(third) {
+        // The ActorSystem continues running and will reply with Quarantined when receiving heartbeats from first,
+        // which result in that first is downing itself, which is undesired.
+        // Normally the ActorSystem will be terminated (or exit JVM) but there could be race conditions during
+        // shutdown that would result in same issue.
+        cluster.shutdown()
```

Comment:
I think we need to look into if there could be some "normal" shutdown scenario when it would send the Artery `Quarantined` message from a node that is not trustworthy.

Reply:
The reason this works with Artery, but it didn't with classic, is that we have a `harmless` flag to `quarantine`.

Log of the harmless quarantine:
```
[info] [JVM-2] [INFO] [11/02/2020 12:55:14.004] [DowningWhenOtherHasQuarantinedThisActorSystem-akka.actor.internal-dispatcher-25] [Association(akka://DowningWhenOtherHasQuarantinedThisActorSystem)] Association to [akka://DowningWhenOtherHasQuarantinedThisActorSystem@localhost:53089] having UID [-663280180320468705] has been stopped. All messages to this UID will be delivered to dead letters. Reason: Cluster member removed, previous status [Up]
```
and then it doesn't send the `Quarantined` message at that point

Reply:
I wonder if a scenario where cluster is combined with remoting could be problematic. Should SBR check if the remote node signalling quarantine is actually a apart of the cluster before killing itself?

Reply:
I think that could be a good precaution.

Thinking about how a normal cluster with "invalid" sbr decisions would behave in that case.
Let's say 4 nodes. Split 2 / 2 and both sides decides to remain for some (invalid) reason. Like auto-downing. Would they then send `Quarantined` to each other if the partition heals and they try to talk to each other?

Reply:
Added a check that the node is part of the cluster (not checking origin member status though, thought about that for a while but couldn't figure out if I should). 

I'll try to cause such a split brain to see what happens.

Reply:
@patriknw I've created a multiJVM test to look at logs and see what happens in such a scenario but I cannot make sense of it, it looks like that even with a test transport blackhole between the two partitions the quarantined message reaches the other side. I can't make sense of it, do you think you could have a look when you have time? The test is in `akka.cluster.SplitBrainQuarantineSpec`.

Reply:
ok, will take a look (during the week)

Reply:
@johanandren I see what you mean. See `def outboundControl`
```
.via(outboundTestFlow(outboundContext))
.viaMat(new OutboundControlJunction(outboundContext, outboundEnvelopePool))(Keep.right)
```

That means that the control messages, such as `Quarantined` are injected into the stream after the outboundTestFlow, which handles the outgoing blackhole.

I don't know if we had some reason for that, but you can try to switch order of those two.

Reply:
Aha, thanks for figuring that one out!

Reply:
After changing the `outboundControl`, I expected split brain/blackhole a 4 node cluster into 2, each downing the other side and then reconnecting after a while would lead to both sides shutting down because they are quarantined by the other side but this did not work.

Reply:
I'll take a look

Reply:
Note that I never pushed that change

Reply:
I tested with the changed order of outboundTestFlow and seems good, and makes more sense. I think we should make that change.